modules/tinymce/src/core/main/ts/api/EditorManager.ts

```diff
@@ -89,6 +90,7 @@ const purgeDestroyedEditor = (editor: Editor | null): void => {
 
 interface EditorManager extends Observable<EditorManagerEventMap> {
   defaultOptions: RawEditorOptions;
+  resources: EditorResources;
```

Comment:
I don't know how I missed this last time, apologies for not bringing it up earlier.

We might need a better name for this; now that I'm not focused on the skin loader code I was about to say "why do we need `EditorManager.resources` when we already have `tinymce.resource`"

I understand that this one is only for pre-caching, where `tinymce.resource` is caching by URL, but they are _very_ similar concepts particularly now that we overloaded the CSS URL in the skin loader.

We can't simply do `tinymce.resources.load(cssurl)`, because if it's not cached it will attempt to download it as JS. But if we add a new method to check if a resource has loaded, that might replace the need for `EditorManager.resources` 🤔 

I'm not sure which path to suggest here. I would prefer to delete this and find a way to use the existing resource cache, but it might not be an easy change.

- [ ] Rename, or replace with a tweaked`tinymce.resource`

Reply:
The problem I see with that is that the 'outside' resources aren't loaded due to activity from the editor, 'adds themselves' when they are run. They'd need to load themselves to each editor as it boots up ( presumably by listening to events in EditorManager... ) which doesn't seem that practical. `EditorManager.resources` means there's a set target for resources to be loaded into ( and loaded once ).

Reply:
Unless I have missed something. If you have a working idea I'd be glad to see if we can add it.

Reply:
My idea was just to use the existing resource API since skins but maybe add a get function. So the .js would do something like:
tinymce.resources.add('skin/dark/whatever'. 'bunch of css');

The you would have the new get function:
const cssData = tinymce.resouces.get('skin/dark.whatever');
if (cssData) {
   addStyleTagToPage(editor, cssData);
} else {
  loadCssFromUrl(someUrl);
}

I really thought we had a `get` function but I see now that we only have a load that would try to load the resource in from an url if not found. Get could be a sync function since it would not do anything async like loading.

Reply:
Maybe I am just missing something obvious, but how would those call get a link to the editor? The closest I can think of would be to use `tinymce.activeEditor`, but that'd leave all other instances in the dirt. Or is `tinymce.resources` an instance shared between all editors?

Reply:
Yes `tinymce.resource` is a global pool shared across the entire page, the same way your `EditorManager.resources` is. The `tinymce.resource.load` api requires a URL, but `tinymce.resource.add` doesn't - it just uses a string key.

Spocke and I both believe that if you make a small tweak to the existing resource pool - I agree with his suggestion of adding the `get` method - it will be equivalent to the resource pool you created.

At that point it would be a more or less direct translation of the code you have today to the global resource api; continue using the CSS URL as the key, and it will be up to the bundle code to make sure that URL is added as a string key to the global resource pool before the editor loads.

Reply:
Alright.